src/uniformsort.hpp

```diff
@@ -44,18 +36,19 @@ namespace UniformSort {
         uint64_t const num_entries,
         uint32_t const bits_begin)
     {
-        uint64_t const memory_len = Util::RoundSize(num_entries) * entry_len;
+        uint64_t const rounded_entries = Util::RoundSize(num_entries);
         auto const swap_space = std::make_unique<uint8_t[]>(entry_len);
         auto const buffer = std::make_unique<uint8_t[]>(BUF_SIZE);
         uint64_t bucket_length = 0;
         // The number of buckets needed (the smallest power of 2 greater than 2 * num_entries).
         while ((1ULL << bucket_length) < 2 * num_entries) bucket_length++;
-        memset(memory, 0, memory_len);
+
+        auto const is_used = std::make_unique<uint8_t[]>(rounded_entries);
```

Comment:
it seems you only use 1 out of these 8 bits. You can make this use 1/8th the amount of memory.
You could use the bitfield class we already have: https://github.com/Chia-Network/chiapos/blob/main/src/bitfield.hpp

Reply:
Ah, yes, that's much nicer. There's a tiny speed hit (back up to 365sec), but that's okay I think.

Reply:
Well, maybe hold off on this as POPCNT is not supported by all CPUs and we use this even on non bitfield

Reply:
this patch doesn't use the bitfield to count bits though.

Reply:
but it's a good point I hadn't thought of. You can't use all parts of the bitfield class unconditionally